lib/versioning/npm/index.spec.ts

```diff
@@ -1,364 +1,98 @@
 import { api as semver } from '.';
 
 describe('versioning/npm/index', () => {
-  describe('semver.isValid(input)', () => {
-    it('should return null for irregular versions', () => {
-      expect(semver.isValid('17.04.0')).toBeFalsy();
-    });
-    it('should support simple semver', () => {
-      expect(semver.isValid('1.2.3')).toBeTruthy();
-    });
-    it('should support semver with dash', () => {
-      expect(semver.isValid('1.2.3-foo')).toBeTruthy();
-    });
-    it('should reject semver without dash', () => {
-      expect(semver.isValid('1.2.3foo')).toBeFalsy();
-    });
-    it('should support ranges', () => {
-      expect(semver.isValid('~1.2.3')).toBeTruthy();
-      expect(semver.isValid('^1.2.3')).toBeTruthy();
-      expect(semver.isValid('>1.2.3')).toBeTruthy();
-    });
-    it('should reject github repositories', () => {
-      expect(semver.isValid('renovatebot/renovate')).toBeFalsy();
-      expect(semver.isValid('renovatebot/renovate#master')).toBeFalsy();
-      expect(
-        semver.isValid('https://github.com/renovatebot/renovate.git')
-      ).toBeFalsy();
-    });
+  test.each`
+    version                                          | isValid
+    ${'17.04.0'}                                     | ${false}
+    ${'1.2.3'}                                       | ${true}
+    ${'1.2.3-foo'}                                   | ${true}
+    ${'1.2.3foo'}                                    | ${false}
```

Comment:
Why is this type of wrapping necessary?

Reply:
It's using Template Literals. It's more readable than the object array version of `test.each`

https://jestjs.io/docs/api#2-testeachtablename-fn-timeout
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Reply:
Despite of ugly `${'...'}`, it's also easier to write them comparing to arrays, and it's supported by Prettier